metrics/local_test.go

```diff
@@ -10,11 +9,6 @@ import (
 )
 
 func TestLocalMetrics(t *testing.T) {
-	numGoroutines := runtime.NumGoroutine()
-	defer func() {
-		assert.Equal(t, numGoroutines, runtime.NumGoroutine(), "Leaked at least one goroutine.")
-	}()
```

Comment:
Looks like you're correct. The `stop()` function waits until the bg routine that's being stopped writes to the `stopped` channel, and then `stop()`  exists. But the bg routine may still linger and trip the count.

We could do the test differently, without actually testing that the go-routine died, but only testing that the function has exited, e.g. by flipping an atomic var or a waitgroup.